src/aws/flb_aws_util.c

```diff
@@ -309,6 +310,12 @@ struct flb_http_client *request_do(struct flb_aws_client *aws_client,
         goto error;
     }
 
+    /* Set max response buffer size for the http client. 
+     * Try setting resp.max_size to 0 to allow buffer to be resized. Zero means unlimited capacity as per requirement. 
+     * Whenever the buffer gets filled out, FluentBit increases it by 32KB in each request/iteration.
+     */
+    flb_http_buffer_size(c, MAX_AWS_RESP_BUFFER_SIZE);
```

Comment:
Comment is too verbose. We can remove the part about increasing it by 32 kb since that is based on current code and can change. I'd make this `Increase the maximum HTTP response buffer size to fit large responses from AWS services`

Reply:
We should also print a warning if it fails.